src/main/fc/rc_modes.c

```diff
@@ -108,13 +113,8 @@ void updateActivatedModes(void)
     bitArraySet(&stickyModes, BOXPARALYZE);
 
     // determine which conditions set/clear the mode
-    for (int i = 0; i < MAX_MODE_ACTIVATION_CONDITION_COUNT; i++) {
-        const modeActivationCondition_t *mac = modeActivationConditions(i);
-
-        // Skip linked macs for now to fully determine target states
-        if (mac->linkedTo) {
-            continue;
-        }
+    for (int i = 0; i < activeMacCount; i++) {
```

Comment:
this would become:
```
for (macMask_t mask = activeMacMask; mask; mask &= mask - 1) {
   const modeActivationCondition_t *mac = modeActivationConditions(builtin_ctz(mask));